handlers/agentinfo/info_test.go

```diff
@@ -233,3 +236,13 @@ func TestIsUsageDataEnabled(t *testing.T) {
 	t.Setenv(envconfig.CWAGENT_USAGE_DATA, "FALSE")
 	assert.False(t, getUsageDataEnabled())
 }
+
+func TestSharedConfigFallback(t *testing.T) {
+	defer sharedConfigFallback.Store(false)
+	assert.Nil(t, getSharedConfigFallback())
+	RecordSharedConfigFallback()
+	assert.Equal(t, 1, *(getSharedConfigFallback()))
+	RecordSharedConfigFallback()
+	RecordSharedConfigFallback()
+	assert.Equal(t, 1, *(getSharedConfigFallback()))
```

Comment:
I'm not sure how we are trying to record the fallback counts. Is this supposed to be 2?

Reply:
It's a boolean, so no matter how many times RecordSharedConfigFallback is called, it'll always be 1. We convert it into an int because it's intended to be a metric.